### PR TITLE
Update supported versions in README.md

### PR DESCRIPTION
**Pull request by Frassle**
# Description

Update the python and go versions to match the currently supported minimum versions.

@AaronFriel pointed out to me the other day that Golangs supported version text is a bit hard to parse but it means only the latest two releases are supported (i.e. 1.18 and 1.19).

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Review by AaronFriel — APPROVED** [review]

**Comment by Frassle** [comment]
Bors merge

**Review by susanev — APPROVED** [review]

**Comment by Frassle** [comment]
bors retry

**Comment by Frassle** [comment]
bors retry
